tests/cbmc/stubs/s2n_libcrypto_is_awslc.c

```diff
@@ -0,0 +1,31 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+* this file except in compliance with the License. A copy of the License is
+* located at
+*
+*     http://aws.amazon.com/apache2.0/
+*
+* or in the "license" file accompanying this file. This file is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+* implied. See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include <cbmc_proof/nondet.h>
+
+#include <stdbool.h>
+#include "crypto/s2n_openssl.h"
+
+static int flag = 0;
+static bool s2n_awslc_flag = 0;
+
+bool s2n_libcrypto_is_awslc()
```

Comment:
What happens if you don't stub this method? It's always false?

Reply:
Yeah, if we don't stub it, it should always return false since it checks `OPENSSL_IS_AWSLC`, which shouldn't be defined in CBMC:
https://github.com/aws/s2n-tls/blob/8ee7a3cb07dbd5020310023ad12e2acb151f1b09/crypto/s2n_libcrypto.c#L114

So if we didn't want to test the awslc part in CBMC we should be fine having it just always return false. I [tested removing the stub](https://github.com/aws/s2n-tls/actions/runs/7657186541/job/20867275714?pr=4361), but `s2n_libcrypto_is_awslc()` is in `s2n_libcrypto.c`, which contains libcrypto symbols like `SSLeay_version`. So in order to include this file, we'd have to stub these symbols out anyway. So I think it probably makes sense to just stub `s2n_libcrypto_is_awslc()` itself.